<?xml version="1.0"?>

<!-- map.launch -->
<launch>
    <!-- Waypoints -->
    <arg name="simulation" default="false"/>
    <include file="$(find igvc_navigation)/launch/waypoint_source.launch"
             unless="$(arg simulation)"/>

    <!-- Planner -->
    <node name="path_planner" pkg="igvc_navigation" type="d_lite_planner" output="screen" >
        <!-- Distance at which a node is considered a valid goal -->
        <param name="goal_range" type="double" value="0.3"/>
        <!-- Robot configuration space -->
        <param name="c_space" type="double" value="0.5"/>
        <!-- Path calculation rate -->
        <param name="rate" type="double" value="10"/>
        <!-- Max distance to a goal node -->
        <param name="maximum_distance" type="double" value="100.0"/>
        <!-- Publish a pointcloud of search nodes on the /expanded topic -->
        <param name="publish_expanded" type="bool" value="true"/>
        <!-- If, for some reason, the path planner decides no optimal path
        currently exists, use the previously generated path instead -->
        <param name="follow_old_path" type="bool" value="true"/>
    </node>

    <!-- Follower -->
    <node name="path_follower" pkg="igvc_navigation" type="path_follower" output="screen" >
        <!-- Maxmimum velocity allowed before stopping motors -->
        <param name="maximum_vel" type="double" value="2.0"/>
        <param name="axle_length" type="double" value="0.52"/>
        <!-- Distance along the path to generate smooth control law for  -->
        <param name="lookahead_dist" type="double" value="1.3"/>
        <!-- granularity with which to visualize planned trajectories -->
        <param name="granularity" type="int" value="100"/>
        <!-- How far away robot should be from the waypoint before it stops -->
        <param name="stop_dist" type="double" value="0.85"/>
        <!-- Smooth Control Parameters -->
        <param name="target_v" type="double" value="0.4"/>
        <param name="k1" type="double" value="1.8"/>
        <param name="k2" type="double" value="5"/>
        <param name="debug" type="bool" value="true"/>
    </node>

</launch>
